src/Microsoft.Extensions.DependencyInjection.Abstractions/FactoryServiceDescriptor.cs

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Diagnostics;
+
+namespace Microsoft.Extensions.DependencyInjection
+{
+    [DebuggerDisplay("Lifetime = {Lifetime}, ServiceType = {ServiceType}, ImplementationFactory = {ImplementationFactory}")]
+    public class FactoryServiceDescriptor : ServiceDescriptor
+    {
+        /// <summary>
+        /// Initializes a new instance of <see cref="FactoryServiceDescriptor"/> with the specified <paramref name="factory"/>.
+        /// </summary>
+        /// <param name="serviceType">The <see cref="Type"/> of the service.</param>
+        /// <param name="factory">A factory used for creating service instances.</param>
+        /// <param name="lifetime">The <see cref="ServiceLifetime"/> of the service.</param>
+        public FactoryServiceDescriptor(Type serviceType, Func<IServiceProvider, object> factory,
+            ServiceLifetime lifetime)
+            : base(serviceType, lifetime)
+        {
+            if (serviceType == null)
```

Comment:
Is there a reason you're checking this in some descriptors, but not in `EnumerableServiceDescriptor` and `OrderedEnumerableServiceDescriptor`? Shouldn't this`null` check just be moved up to the base class? I don't believe you ever want `ServiceType` to be `null`....